resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1005,26 +1005,6 @@ private[spark] class Client(
     val tmpDir = new Path(Environment.PWD.$$(), YarnConfiguration.DEFAULT_CONTAINER_TEMP_DIR)
     javaOpts += "-Djava.io.tmpdir=" + tmpDir
 
-    // TODO: Remove once cpuset version is pushed out.
-    // The context is, default gc for server class machines ends up using all cores to do gc -
-    // hence if there are multiple containers in same node, Spark GC affects all other containers'
-    // performance (which can be that of other Spark containers)
-    // Instead of using this, rely on cpusets by YARN to enforce "proper" Spark behavior in
-    // multi-tenant environments. Not sure how default Java GC behaves if it is limited to subset
-    // of cores on a node.
-    val useConcurrentAndIncrementalGC = launchEnv.get("SPARK_USE_CONC_INCR_GC").exists(_.toBoolean)
```

Comment:
Java 14 starts to remove support for CMS([JEP-363](https://openjdk.org/jeps/363)). So when we use Java 17 to run Spark and set `SPARK_USE_CONC_INCR_GC=true`, CMS will not take effect (No error will be reported due to `-XX:+IgnoreUnrecognizedVMOptions` is configured by default).

It seems that `SPARK_USE_CONC_INCR_GC` is only use by Yarn AppMaster. We can also achieve the same goal by directly configuring `spark.driver.extraJavaOptions(cluster mode)` or `spark.yarn.am.extraJavaOptions(client mode)`. At the same time, I found that `SPARK_USE_CONC_INCR_GC` is not exposed to users in any user documents, so I think we can clean up this environment variable.

But because I lack some context for this env, so I wonder if we can remove it? @srowen @dongjoon-hyun @HyukjinKwon

Reply:
For the YARN code, @mridulm and @tgravescs knows better~

Reply:
Thanks @dongjoon-hyun

Reply:
yeah this is pretty old, not sure if anyone is using it anymore. Personally think its ok to remove since users can set themselves.

Reply:
Thanks @tgravescs , I will make this one to a formal pr later

Reply:
Agree, this is pretty old ... I dont think there are a lot of usages of this now.
Should be fine to remove - and if there are concerns, users can explicitly add it anyway - like Tom suggested.